### PR TITLE
[doc&tools] librispeech recipe

### PR DESCRIPTION
**Pull request by whiteshirt0429**

### CODE REVIEW

File: examples/librispeech/s0/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+On conformer without speed perpurb and lm
+* configure: conf/train_conformer_large.yaml
+* beam size: 10
+* gpu nums: 8
+* ctc weight(used for attention rescoring): 0.5
+
+test clean
+| decoding mode/chunk size | full |
+|--------------------------|------|
+| attention rescoring      | 2,85 |
```
Comment (robin1001) [inline examples/librispeech/s0/README.md:14]: 2.85

File: examples/librispeech/s0/conf/train_conformer_large.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    normalize_before: true
+    cnn_module_kernel: 31
+    use_cnn_module: True
+    activation_type: 'swish'
+    pos_enc_layer_type: 'rel_pos'
+    selfattention_layer_type: 'rel_selfattn'
+
+# decoder related
+decoder: transformer
+decoder_conf:
```
Comment (robin1001) [inline examples/librispeech/s0/conf/train_conformer_large.yaml:22]: whether the output size should be modified to 512 for a large conformer?

File: examples/librispeech/s0/conf/train_conformer_large.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+# hybrid CTC/attention
+model_conf:
+    ctc_weight: 0.3
+    lsm_weight: 0.1     # label smoothing option
+    length_normalized_loss: false
+
+
+collate_conf:
+    spec_aug: true
+    norm_mean: true # if true, must specify cmvn file in cmd line
```
Comment (robin1001) [inline examples/librispeech/s0/conf/train_conformer_large.yaml:40]: what's the default configure in espnet for cmvn, spec augmentation?

File: examples/librispeech/s0/conf/train_conformer.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
+# network architecture
```
Comment (robin1001) [inline examples/librispeech/s0/conf/train_conformer.yaml:1]: remove this if you don't have any experiment result on this recipe

File: examples/librispeech/s0/conf/train_transformer.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
+# network architecture
```
Comment (robin1001) [inline examples/librispeech/s0/conf/train_transformer.yaml:1]: remove this

File: examples/librispeech/s0/conf/train_unified_conformer.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
+# network architecture
```
Comment (robin1001) [inline examples/librispeech/s0/conf/train_unified_conformer.yaml:1]: remove this or add it later

File: examples/librispeech/s0/run.sh [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+export CUDA_VISIBLE_DEVICES="0,1,2,3,4,5,6,7"
+stage=0 # start from 0 if you need to start from data preparation
+stop_stage=5
+# data
+data_url=www.openslr.org/resources/12
+# use your own data path
+datadir=/nfsa/diwu/open-dir
+nj=16
+dict=data/dict/lang_char.txt
+# Optional train_config
```
Comment (robin1001) [inline examples/librispeech/s0/run.sh:19]: refine this comment or remove this

File: examples/librispeech/s0/run.sh [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+# just 1gpu, otherwise it's is multiple gpu training based on DDP in pytorch
+export CUDA_VISIBLE_DEVICES="0,1,2,3,4,5,6,7"
+stage=0 # start from 0 if you need to start from data preparation
+stop_stage=5
+# data
+data_url=www.openslr.org/resources/12
+# use your own data path
+datadir=/nfsa/diwu/open-dir
+nj=16
+dict=data/dict/lang_char.txt
```
Comment (robin1001) [inline examples/librispeech/s0/run.sh:18]: conflicts with the dict declared below

File: examples/librispeech/s0/run.sh [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+    # -1 for full chunk
+    decoding_chunk_size=
+    ctc_weight=0.5
+    for test in $recog_set; do 
+    for mode in attention_rescoring ctc_greedy_search ctc_prefix_beam_search attention; do
+    {
+        test_dir=$dir/${test}_${mode}
+        mkdir -p $test_dir
+        python wenet/bin/recognize.py --gpu 0 \
+            --mode $mode \
```
Comment (robin1001) [inline examples/librispeech/s0/run.sh:184]: indent here

File: requirements.txt [hunk 10]
```diff
@@ -2,3 +2,5 @@ flake8
 pyyaml
 tensorboardX
 typeguard
+Pillow
```
Comment (robin1001) [inline requirements.txt:5]: where Pillow is used?

File: tools/spm_decode [hunk 11]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
```
Comment (robin1001) [inline tools/spm_decode:1]: what's the benefits of wrapping spm_encode/spm_decode/spm_train? 
If just for easier input output interface, the raw spm_* are recommended.

File: requirements.txt (hunk 10 above)
Comment (whiteshirt0429) [reply requirements.txt:5]: spec aug time warp， will be added

**Review by robin1001 — APPROVED** [review]

**Pull request merged**